### PR TITLE
feat: Add _tag and _old,_new grouping to extr serializer

### PR DESCRIPTION
**Pull request by arnelplim**
extr serializer changes
- Add support for _tag and tags grouping.
- Add support for _old,_new grouping IDs

### CODE REVIEW

File: plugins/serializers/extr/README.md [hunk 1]
```diff
@@ -2,14 +2,19 @@
...
+        ifAdminStatus_old=1,ifAdminStatus_new=1
+        --> "ifAdminStatus":{"old":1,"new":0}
    - Groups metric fields appended with _key.
-       ifIndex_key=1, name_key="1:2"
-       --> "key":{ifIndex:1, name:"1:2:}
+        ifIndex_key=1, name_key="1:2"
+        --> "key":{ifIndex:1, name:"1:2:}
+   - Groups metric fields appended with _tag.
+        partNumber_tag="1647G-00129 800751-00-01", revision_tag="01"
+        --> "tags":{ifIndex:"1647G-00129 800751-00-01", revision:"01"}
```
Comment (vullanatt) [inline plugins/serializers/extr/README.md:15]: ifIndex should be partNumber_tag on this line?

File: plugins/serializers/extr/README.md [hunk 2]
```diff
@@ -2,14 +2,19 @@
 
 The `extr` output data format converts metrics into JSON documents, performing the following operatins on batched metrics:
    - Combines sequential metrics matching name, tags, and timestamps into a single JSON metric, combining the fields of each metric into an array of fields.
-   - Groups metric fields appended with _min, _max, _avg
+   - Groups metric fields appended with _min, _max, _avg or _old, _new
         usage_min=1,usage_max=100,usage_avg-50
```
Comment (vullanatt) [inline plugins/serializers/extr/README.md:6]: usage_avg=50

File: plugins/serializers/extr/README.md [hunk 3]
```diff
@@ -2,14 +2,19 @@
...
         --> "usage":{"avg":50,"max":100,"min":1}
+        ifAdminStatus_old=1,ifAdminStatus_new=1
+        --> "ifAdminStatus":{"old":1,"new":0}
    - Groups metric fields appended with _key.
-       ifIndex_key=1, name_key="1:2"
-       --> "key":{ifIndex:1, name:"1:2:}
+        ifIndex_key=1, name_key="1:2"
+        --> "key":{ifIndex:1, name:"1:2:}
```
Comment (vullanatt) [inline plugins/serializers/extr/README.md:12]: Is there a double quote " mismatch around 1:2:

File: plugins/serializers/extr/README.md [hunk 4]
```diff
@@ -50,16 +55,20 @@ The `extr` output data format converts metrics into JSON documents, performing t
...
-CpuStats,serialnum=XYZ-1234 core_key=1i,usage_min=22.2,usage_max=89.2,usage_avg=41.2
-CpuStats,serialnum=XYZ-1234 core_key=2i,usage_min=33.2,usage_max=79.2,usage_avg=47.2
-FanStats,serialnum=XYZ-1234 slot_key=1i,tray_key=2i,fan_key=10i,rpm_min=4112,rpm_max=5012,rpm_avg=4212,pwm_min=32,pwm_max=52,pwm_avg=32
-FanStats,serialnum=XYZ-1234 slot_key=1i,tray_key=2i,fan_key=11i,rpm_min=5002,rpm_max=5092,rpm_avg=4102,pwm_min=52,pwm_max=62,pwm_avg=52
+CpuStats,serialnumber=XYZ-1234 core_key=0i,usage_min=35.1,usage_max=99.1,usage_avg=35.1
+CpuStats,serialnumber=XYZ-1234 core_key=1i,usage_min=50.1,usage_max=88.1,usage_avg=51.1
+FanStateChanged,serialnumber=XYZ-1234 slot_key=2i,tray_key=4i,fan_key=1i,partNumber_tag="1647G-00129 800751-00-01",revision_tag="01",airFlowDirection_tag="FrontToBack",state_old="Ok",state_new="Failed"
+FanStateChanged,serialnumber=XYZ-1234 slot_key=2i,tray_key=4i,fan_key=2i,partNumber_tag="1647G-00129 800751-00-01",revision_tag="01",airFlowDirection_tag="FrontToBack",state_old="Failed",state_new="Ok"
+FanStats,serialnumber=XYZ-1234 slot_key=1i,tray_key=2i,fan_key=10i,rpm_min=4101,rpm_max=5001,rpm_avg=4201,pwm_min=31,pwm_max=41,pwm_avg=31
```
Comment (vullanatt) [inline plugins/serializers/extr/README.md:62]: slot_key is removed now. In lines 62, 63 and 64.

File: plugins/serializers/extr/README.md [hunk 5]
```diff
@@ -50,16 +55,20 @@ The `extr` output data format converts metrics into JSON documents, performing t
...
+FanStats,serialnumber=XYZ-1234 slot_key=1i,tray_key=2i,fan_key=11i,rpm_min=4001,rpm_max=4991,rpm_avg=4001,pwm_min=41,pwm_max=51,pwm_avg=41
+FanStats,serialnumber=XYZ-1234 slot_key=2i,tray_key=3i,fan_key=9i,rpm_min=2101,rpm_max=3211,rpm_avg=2201,pwm_min=11,pwm_max=41,pwm_avg=11
+InterfaceStateChanged,serialnumber=XYZ-1234 ifIndex_key=1001,name_key="1:1",adminStatus_old=0,adminStatus_new=1,operStatus_old=0,operStatus_new=1
+InterfaceStateChanged,serialnumber=XYZ-1234 ifIndex_key=1002,name_key="1:2",adminStatus_old=0,adminStatus_new=1,operStatus_old=0,operStatus_new=1
+CpuStats,serialnumber=XYZ-1234 core_key=0i,usage_min=10.2,usage_max=91.2,usage_avg=44.2
+CpuStats,serialnumber=XYZ-1234 core_key=1i,usage_min=22.2,usage_max=89.2,usage_avg=41.2
+CpuStats,serialnumber=XYZ-1234 core_key=2i,usage_min=33.2,usage_max=79.2,usage_avg=47.2
+FanStats,serialnumber=XYZ-1234 slot_key=1i,tray_key=2i,fan_key=10i,rpm_min=4112,rpm_max=5012,rpm_avg=4212,pwm_min=32,pwm_max=52,pwm_avg=32
```
Comment (vullanatt) [inline plugins/serializers/extr/README.md:70]: slot_key can be removed in lines 70 and 71 also.

File: plugins/serializers/extr/README.md [hunk 6]
```diff
@@ -133,13 +142,56 @@ will serialize into the following extr JSON ouput
...
+  "fanStateChanged": [
+    {
+      "device": {
+        "serialnumber": "XYZ-1234"
+      },
+      "items": [
+        {
+          "keys": {
+            "fan": 1,
+            "slot": 2,
```
Comment (vullanatt) [inline plugins/serializers/extr/README.md:157]: slot can be removed

File: plugins/serializers/extr/README.md [hunk 7]
```diff
@@ -133,13 +142,56 @@ will serialize into the following extr JSON ouput
...
+          "tags": {
+            "airFlowDirection": "FrontToBack",
+            "partNumber": "1647G-00129 800751-00-01",
+            "revision": "01"
+          }
+        },
+        {
+          "keys": {
+            "fan": 2,
+            "slot": 2,
```
Comment (vullanatt) [inline plugins/serializers/extr/README.md:173]: ditto

**Review by vullanatt — APPROVED** [review]

File: plugins/serializers/extr/README.md [hunk 8]
```diff
@@ -238,7 +290,46 @@ will serialize into the following extr JSON ouput
...
+    }
+  ],
+  "interfaceStateChanged": [
+    {
+      "device": {
+        "serialnumber": "XYZ-1234"
+      },
+      "items": [
+        {
+          "adminStatus": {
```
Comment (dshnayder) [inline plugins/serializers/extr/README.md:303]: According to OpenAPI, adminStatus and operStatus are enums:
```yaml
      properties:
        adminStatus:
          enum: 
            - Down 
            - Up
        operStatus:
          enum:
            - Down
            - Up
```

**Review by dshnayder — APPROVED** [review]

File: plugins/serializers/extr/README.md (hunk 3 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:12]: I'll fix.

File: plugins/serializers/extr/README.md (hunk 1 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:15]: I'll fix

File: plugins/serializers/extr/README.md (hunk 4 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:62]: I'll fix.  Note, this is just example output, so not a big deal.

File: plugins/serializers/extr/README.md (hunk 2 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:6]: Fixed

File: plugins/serializers/extr/README.md (hunk 3 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:12]: Fixed

File: plugins/serializers/extr/README.md (hunk 4 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:62]: Fixed

File: plugins/serializers/extr/README.md (hunk 6 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:157]: Fixed

File: plugins/serializers/extr/README.md (hunk 7 above)
Comment (arnelplim) [reply plugins/serializers/extr/README.md:173]: Fixed

**Comment by arnelplim** [comment]
I had already pushed my changes before dealing with Venu's comments.  I'll go ahead and push those corrections directly into extr-1.21 since this Pull Request is already closed.

**Pull request merged**